app/knight.py

```diff
@@ -0,0 +1,62 @@
+from typing import Optional
+
+from app.items import Weapon, Armour, Potion
+
+
+class Knight:
+    def __init__(
+        self,
+        name: str,
+        power: int,
+        hp: int,
+        weapon: dict,
+        armour: list[dict],
+        potion: Optional[dict] = None,
+    ) -> None:
+        self.name = name
+        self._base_power = power
+        self.power = self._base_power

```

Comment:
I do not understand why we need both properties. As for me, it is redundant code. It makes code more unreadable and harder

Reply:
The same for each attribute